scripts/tests/missions/bastok.lua

```diff
@@ -0,0 +1,171 @@
+describe('Bastok Missions', function()
+    ---@type CTestPuppet
+    local puppet
+
+    before_each(function()
+        puppet = xi.test.world:spawnPlayer()
+
+        puppet:setNation(xi.nation.BASTOK)
+
+        -- SOA first cutscene conflicts
+        puppet:addMission(xi.mission.log_id.SOA, xi.mission.id.soa.RUMORS_FROM_THE_WEST)
+        local missionStatus = puppet:getMissionStatus(xi.mission.log_id.SOA)
+        missionStatus = utils.mask.setBit(missionStatus, 0, true)
+        missionStatus = utils.mask.setBit(missionStatus, 1, true)
+        puppet:setMissionStatus(xi.mission.log_id.SOA, missionStatus)
+        puppet:completeMission(xi.mission.log_id.SOA, xi.mission.id.soa.RUMORS_FROM_THE_WEST)
+    end)
+
+    describe('The Zeruhn Report', function()
+        it('should complete the mission', function()
+            puppet:setRank(1)
+
+            -- Pick up mission from gate guard.
+            puppet:gotoZone(xi.zone.BASTOK_MARKETS)
+            puppet.entities:gotoAndTrigger('Cleades', { eventId = 1000, finishOption = 0 })
```

Comment:
I would expect this to be: `puppet:gotoAndTrigger(...`, with a more obvious verify/expects payload. I haven't looked at how this is laid out/bound yet though

Reply:
Or even:

puppet:gotoAndTrigger('Name') -> returns the table

```lua
assertEQ(puppet:gotoAndTrigger('Name'), { eventId = 120 })
```